### PR TITLE
Add support for GNOME 40

### PR DESCRIPTION
**Pull request by frandieguez**
Hi,

please consider merging these changes. I have updated the prefs panel to support GNOME 40

Regards

### CODE REVIEW

File: prefs.js [hunk 1]
```diff
@@ -226,7 +229,8 @@ const App = new Lang.Class({
         SettingsSchema.bind(Fields.STRIP_TEXT, this.field_strip_text, 'active', Gio.SettingsBindFlags.DEFAULT);
         SettingsSchema.bind(Fields.ENABLE_KEYBINDING, this.field_keybinding_activation, 'active', Gio.SettingsBindFlags.DEFAULT);
 
-        this.main.show_all();
+        if (this.main.show_all)
```
Comment (rmader) [inline prefs.js:232]: Maybe `typeof this.main.show_all === 'function'`?

File: metadata.json [hunk 2]
```diff
@@ -1,8 +1,7 @@
 {
     "shell-version": [
         "3.38",
-        "40.beta",
-        "40.rc"
+        "40.0"
```
Comment (rmader) [inline metadata.json:4]: Maybe just use `40` and squash the commit? That's what's done for the officially supported ones at https://gitlab.gnome.org/GNOME/gnome-shell-extensions/

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,27 @@
+PREFIX ?= ~/.local/share/gnome-shell/extensions
+
+TRANSLATION_ID=clipboard-indicator
+EXTENSION_UUID=clipboard-indicator@tudmotu.com
+
+all: build install
```
Comment (rmader) [inline Makefile:6]: I'd drop the `install`, assuming most people do not expect a `install` when just running `make`. Also, maybe move this to a own commit?

**Comment by frandieguez** [comment]
Hi! I'll fix all the issues you've find and fix them.

**Comment by rmader** [comment]
> Hi! I'll fix all the issues you've find.

Thanks! Concerning the Makefile: maybe we want to go with https://github.com/Tudmotu/gnome-shell-extension-clipboard-indicator/pull/234 instead as it does a bit more than the one here.

**Comment by frandieguez** [comment]
> > Hi! I'll fix all the issues you've find.
> 
> Thanks! Concerning the Makefile: maybe we want to go with #234 instead as it does a bit more than the one here.

yes, I'll revert that change. I like the way it works the other Makefile, as it goes all the way porting from .sh files to final Makefile, which is what I'll like to do in the first place.

File: metadata.json (hunk 2 above)
Comment (rmader) [reply metadata.json:4]: FTR, I'd recommend using really just `40` instead of `40.0`

Comment (rmader) [reply metadata.json:4]: To explain: from 40 on the major version signals API stability - independent of the point release or beta/release candidate. And API stability is everything we are interested in as extension.

Comment (frandieguez) [reply metadata.json]: Uploading new version, right now.

**Review by frandieguez — COMMENTED** [review]
I think that it's all fixed. Please tell me if you need more changes

**Comment by prateekmedia** [comment]
Please also get rid of dropdown button as it looks ugly with gnome 40 design and also because they removed it from the power and network strip.

**Comment by prateekmedia** [comment]
Also one thing to notice is that the keyboard shortcuts inside the preferences says undefined but we can still set it, can you look into it please.

File: prefs.js [hunk 4]
```diff
@@ -183,8 +186,8 @@ const App = new Lang.Class({
                 let inputWidget = input;
 
                 if (input instanceof Gtk.Switch) {
-                    inputWidget = new Gtk.HBox();
-                    inputWidget.pack_end(input, false, false, 0);
+                    inputWidget = new Gtk.Box();
+                    inputWidget.append(input);
```
Comment (prateekmedia) [inline prefs.js:190]: Append would not work in 3.38, either you should check first that is it valid, so that you could use pack_start for 3.38, or else you should remove 3.38 support.

**Review by prateekmedia — CHANGES_REQUESTED** [review]

File: prefs.js (hunk 4 above)
Comment (frandieguez) [reply prefs.js]: Should be fixed

**Comment by frandieguez** [comment]
Please check https://github.com/Tudmotu/gnome-shell-extension-clipboard-indicator/pull/280 that PR is more complete

**Comment by Tudmotu** [comment]
@frandieguez Thank you for taking the time to send this PR, I really appreciate it!
But I'm gonna go with #280 so I am closing this one. But really - thank you for opening this PR as well :pray:
